nrf: Enable micro:bit FS by default

glennrub opened the pull request:
Update configuration define from
MICROPY_HW_HAS_BUILTIN_FLASH to MICROPY_MBFS.

MICROPY_MBFS will enable the builtin flash as
part of enabling the micro:bit FS.

aykevl commented:
All devices will have internal flash, as they're all from the nRF5x family. What this configuration actually does is enabling the microbit filesystem.

What about dropping the `MICROPY_HW_HAS_BUILTIN_FLASH` option for all flash-related stuff and instead using something like `MICROPY_MBFS` just for the filesystem itself?
Code to access flash will be removed when no filesystem code uses it so it's not really necessary to disable it separately.

glennrub commented:
Good point! I'll rework this PR to use a microbit-fs as enabler. I wonder if it would make sense to call it something like "MICROPY_PY_MBFS" and also move the configuration to the top mpconfigport.h? What do you think of that?

glennrub committed:
nrf: Enable micro:bit FS by default

Update configuration define from
MICROPY_HW_HAS_BUILTIN_FLASH to MICROPY_MBFS.

MICROPY_MBFS will enable the builtin flash as
part of enabling the micro:bit FS.

glennrub commented:
@aykevl, i've force-pushed a replacement of the PR. 

I used your proposed name, MICROPY_MBFS, which made a bit more sense as it is a feature to be enabled and not really a micropython module.

aykevl reviewed (approved):
Looks good!
